src/language/LanguageManager.js

```diff
@@ -263,7 +285,7 @@ define(function (require, exports, module) {
      *
      * @return {Language} The language for the provided file type or the fallback language
      */
-    function getLanguageForPath(path, ignoreOverride) {
+    function getLanguageForPath(path, ignoreOverride, rawText) {
```

Comment:
Are we changing the API to _require_ passing the text content?  That's a breaking API change, which has to be done carefully (and there are many callers in the Brackets core code alone that will need to be updated).  OTOH, if it's optional we're inviting inconsistent behavior: some features in Bracekts will pass only the path (treating Bash files as plain text) while others will pass the full text and treat the file as Bash.  Is there any reasonable, performant way around that?

Might be worth thinking through use cases for `getLanguageForPath()` (and/or surveying current callers) to think through what kinds of problems might arise when different parts of Brackets treat the same file as different types.